mozilla_django_oidc_db/setup_configuration/steps.py

```diff
@@ -0,0 +1,53 @@
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed
+
+from mozilla_django_oidc_db.forms import OIDCSetupConfigForm
+from mozilla_django_oidc_db.models import OpenIDConnectConfig
+from mozilla_django_oidc_db.setup_configuration.models import (
+    AdminOIDCConfigurationModel,
+)
+from mozilla_django_oidc_db.utils import create_missing_groups
+
+
+class AdminOIDCConfigurationStep(BaseConfigurationStep[AdminOIDCConfigurationModel]):
+    """
+    Configure admin login via OpenID Connect
+    """
+
+    verbose_name = "Configuration for admin login via OpenID Connect"
+    config_model = AdminOIDCConfigurationModel
+    namespace = "oidc_db_config_admin_auth"
+    enable_setting = "oidc_db_config_enable"
+
+    def execute(self, model: AdminOIDCConfigurationModel) -> None:
+
+        config = OpenIDConnectConfig.get_solo()
+
+        base_model_data = model.model_dump()
+        endpoint_config_data = base_model_data.pop("endpoint_config")
+
+        all_settings = {
+            "sync_groups": config.sync_groups,
+            "oidc_use_nonce": config.oidc_use_nonce,
+            "enabled": True,
+            "claim_mapping": config.claim_mapping,  # JSONFormField widget cannot handle blank values with object schema
+            "sync_groups_glob_pattern": config.sync_groups_glob_pattern,
+            **base_model_data,
+            **endpoint_config_data,
+        }
+
+        if groups := all_settings.get("default_groups"):
+            all_settings["default_groups"] = create_missing_groups(
+                groups, all_settings["sync_groups_glob_pattern"]
```

Comment:
Should it use the glob pattern here? I do not think it did in the original Open Klant setup config but does in the middleware

Reply:
Also I should probably make a test for this

Reply:
If I read this correctly it will always create missing groups, regardless of the value of `sync_groups`? I would either remove `sync_groups` from the model and always set it to True then, or add it as a conditional here to make sure it works the same as in the admin

Reply:
This is an oversite and I assume it should use the sync_group value